operations/src/main/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkService.java

```diff
@@ -0,0 +1,113 @@
+package edu.unc.lib.boxc.operations.impl.download;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.models.AccessGroupSet;
+import edu.unc.lib.boxc.auth.api.services.AccessControlService;
+import edu.unc.lib.boxc.model.api.exceptions.NotFoundException;
+import edu.unc.lib.boxc.model.api.objects.ContentObject;
+import edu.unc.lib.boxc.model.api.objects.FileObject;
+import edu.unc.lib.boxc.model.api.objects.RepositoryObjectLoader;
+import edu.unc.lib.boxc.model.api.objects.WorkObject;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Service that manages downloading all of a Work's FileObjects as a zip file
+ */
+public class DownloadBulkService {
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repoObjLoader;
+    private Path basePath;
+
+    public Path downloadBulk(DownloadBulkRequest request) {
+        var pidString = request.getWorkPidString();
+        var workPid = PIDs.get(pidString);
+        var agentPrincipals = request.getAgent().getPrincipals();
+        aclService.assertHasAccess(
+                "User does not have permissions to view the Work for download",
+                workPid, agentPrincipals, Permission.viewOriginal);
+
+        var zipFilePath = Paths.get(basePath.toString(), getZipFilename(pidString));
+        try {
+            var workObject = repoObjLoader.getWorkObject(workPid);
+            zipFiles(workObject, agentPrincipals, zipFilePath);
+        } catch (Exception e){
```

Comment:
I don't think we actually need this catch, repoObjLoader already throws NotFoundException when the object isn't found, and ObjectTypeMismatchException when its not a work. We could change it to catching IOException since zipFiles throws that, if that occurs then we probably would want to throw a `ServiceException`.